docs/examples.rst

```diff
@@ -4,4 +4,391 @@
+  as a blue triangle, and the obstacles in the world with black squares. When
+  the agent encounters an obstacle, it changes from a blue to a red colour.
+  Also indicated is the current "learning" state of the agent -- the agent
+  is learning when **Explore** is **ON**, and conversely, is not adapting to
+  the reward signals when **Explore** is **OFF**.
+- The bottom right quadrant displays the output of the action selection network
+  (``Thalamus``), as well as the associated spike rasters for each action
+  (move forward: ``actions_0``; turn counter-clockwise: ``actions_1``;
+  turn clockwise: ``actions_2``)
+
+When the simulation is running, the agent should move around the world, as
+shown in the images below. The images below also show the agent moving around
+the world in 3 modes:
+
+   1. Exploring: The agent is constantly using the error signal to learn
+      which is the best action to perform.
+   #. Exploiting: The learning has been disabled after the agent has spent
+      about 15 seconds in the "Exploring" mode. The agent is thus moving
+      about the world using the radar-to-action mapping it had learned in the
+      exploration phase.
+   #. No Learning: The simulation is started with learning disabled. In this
+      scenario, the agent moves forward, encounters a wall, and is unable to
+      recover.
+
+   .. list-table::
+      :widths: 33 33 33
+      :align: center
+
+      * -
+
+         .. do not delete this table row
+
+        -
+        -
+
+      * -
+
+         .. image:: _static/rl_explore.gif
+
+         |nbsp| |nbsp| |nbsp| Exploring
+
+        -
+
+         .. image:: _static/rl_exploit.gif
+
+         |nbsp| |nbsp| |nbsp| Exploiting
+
+        -
+
+         .. image:: _static/rl_no-learn.gif
+
+         |nbsp| |nbsp| |nbsp| No Learning
+
+Changing the World Layout
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Near the top of the file, a section of code defines the layout of the world
+the agent is exploring. The section of code looks something like:
+
+.. rst-class:: compact
+
+   .. code-block:: python
+
+          world_maps = ["""
+      #########
+      #       #
+      #       #
+      #   ##  #
+      #   ##  #
+      #       #
+      #########""",
+
+You can alter the default world map by changing the first entry in the
+``world_maps`` list. Obstacles are indicated with the ``#`` character, and
+unobstructed space is indicated with the space character. The world map can
+be increased or decreased in size by altering the number of rows and columns
+that make up the map.
+
+
+Controlling the Simulation with the Keyboard
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you have the ``keyboard-state`` branch of Nengo GUI you should see
+instructions printed in the console pane on the bottom right. These
+instructions will inform you on which keyboard presses can be used to
+disable / permanently enable learning; reset the agent's location back to the

```

Comment:
Any idea why we have a random keypress here? (^M is enter I believe?)

Reply:
hahaha. no? I didn't see it in my text editor. I'll try to remove it.

Reply:
It doesn't seem to affect anything, but it was weird haha

Reply:
It's a carriage return character, so you probably have a CRLF (Windows) line-ending instead of a LF (Unix) line-ending there.